### PR TITLE
[luci/pass] Support RoPE Operation

### PR DESCRIPTION
**Pull request by ys44kim**
This commit supports RoPE for luci pass

ONE-DCO-1.0-Signed-off-by: youngsik kim ys44.kim@samsung.com

draft : https://github.com/Samsung/ONE/pull/13978
issue : https://github.com/Samsung/ONE/issues/13972

### CODE REVIEW

File: compiler/luci/pass/src/FuseRoPEPass.cpp [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+} // namespace
+
+namespace
+{
+
+bool fuse_rope(luci::CircleAdd *add)
+{
+  RoPEPattern::PatternMode pm = RoPEPattern::PatternMode::Mode_NEOX;
+
+  // TODO support other modes (ex. GPT_J)
```
Comment (ys44kim) [inline compiler/luci/pass/src/FuseRoPEPass.cpp:247]: Currently, only GPT-NEOX is supported, and the default is GPT-NEOX.
If another mode (ex. GPT-J) is added in the future, need to find the pattern and add the code to distinguish the mode.
so this has not been confirmed yet and will be added in the future

File: compiler/luci/pass/src/FuseRoPEPass.cpp [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+ *    - Execution dependency order is top to the bottom.
+ *    - Node name is matched with variable name of RoPEPattern class.
+ *    - Usually, first word of node name (variable name) is node type. For e.g.
+ *      variable 'mean_as_variance' is pointer to TFLMean.
+ *    - (Item in parenthesis) means actually exist, but not having a name and
+ *      not a variable of RoPEPattern class.
+ *
+ *    TODO support other semantically same patterns for RoPE
+ *
+ * NEOX
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseRoPEPass.cpp:45]: maybe?
```suggestion
 * Mode: NEOX
```

File: compiler/luci/pass/src/FuseRoPEPass.cpp [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+#include <set>
+
+// Helper to fuse RoPE(Rotary Position Embedding)
+namespace
+{
+
+/**
+ * SUBGRAPH PATTERN
+ *
+ *    - Below diagram shows Instance Norm pattern to fuse.
```
[the remark this answers is not in the document]
Comment (ys44kim) [reply compiler/luci/pass/src/FuseRoPEPass.cpp]: > maybe this comment needs revision

thanks,
i updated

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**